src/ipc/ipc-zephyr.c

```diff
@@ -125,13 +125,26 @@ enum task_state ipc_platform_do_cmd(struct ipc *ipc)
 	/* perform command */
 	ipc_cmd(hdr);
 
-#if CONFIG_SOC_SERIES_INTEL_ADSP_CAVS
-	/* are we about to enter D3 ? */
-	if (ipc->pm_prepare_D3) {
-		/* no return - memory will be powered off and IPC sent */
-		platform_pm_runtime_power_off();
+	if (ipc->task_mask & IPC_TASK_POWERDOWN) {
+		if (ipc->pm_prepare_D3) {
```

Comment:
Revert to:
```c
if (ipc->pm_prepare_D3)
```

CAVS don't use IPC_TASK_POWERDOWN

Reply:
Well CI tests on Device are green for CAVS platforms. They do use in
`src\platform\intel\cavs\platform.c`
`int platform_context_save(struct sof *sof)`

Reply:
so this file is not used if you don't compile for tgl native drivers. And CI is not doing it yet.

Reply:
I have check it and You are right. But this one IF will do the work here.

Reply:
Removed pm_prepare_d3 if as Zephyr implementation should use IPC masks.
Added second commit with refactoring.
@juimonen I'll check tgl native drivers build manually

Reply:
Ok I've run `python sof\scripts\xtensa-build-zephyr.py tgl -o sof\app\overlays\tgl\ipc4_native_drivers_overlay.conf`
everything builds perfectly fine.